docs/features/role-based-access-control/high-level-design.md

```diff
@@ -0,0 +1,100 @@
+  * user with `Owner` role can perform all the actions a `Contributor` can, as well as deleting actions for the given
+resource (which usually includes calls to `DELETE resource/:id` endpoints).
+
+**However, it is up to each controller method to correctly check permissions for the action being performed.** This also applies
+to custom actions that might be subject to RBAC: for instance, if publishing a project is a custom action that should be
+subject to RBAC, then it is up to the controller action to check that the user performing the request has the required role 
+(e.g. `Owner`) to perform that action.
+
+The mapping between the different platform actions and the role required to perform such action is kept in the following
+spreadsheet: https://docs.google.com/spreadsheets/d/1vz-YHdoAAG2nv_B1QUR7LQ-eU_DnKBexrqULrzMMOtw/edit?usp=sharing
+
+## Association between roles and users for resources
+
+The association between roles and users for resources will be stored in database tables in the `marxan-api` database,
+using one database table for each resource:
+* `users_organizations` will store the roles of users inside organizations.
+* `users_projects` will store the roles of users inside projects.
+* `users_scenarios` will store the roles of users inside scenarios.
+
+These tables should contain columns for `user_id`, `role` and the resource id (`organization_id`, `project_id` and 
+`scenario_id`, respectively).
+
+## Implementation details
+
+In terms of implementation, the existing module `projects-acl` should be renamed to a more generic name such as 
+`access-control` or `rbac` (though I'm not a fan of acronyms as names of modules). This module will be used across 
+the platform as the "single-source-of-truth" to check whether users are allowed to access given resources.
+
+This module, for now, will support checking permissions for the 3 resources subject to RBAC: organizations, projects and
+scenarios. For this, this module should provide 3 services, each of them implementing the following interface:
+
+
+`isOwner(resourceId: string, userId: string): boolean`
+
+Checks if the given user can perform the actions of a `Owner` for the provided resource id. This method should return 
+`true` if one of the following conditions apply:
+
+* the user is a platform admin;
+* the user has a role greater or equal to `Owner` for the resource being checked in the corresponding table (e.g. when 
+ checking if a user is owner of a project, `this.projectAclService.isOwner(projectId, userId)`, there must be a record 
+ in the `users_projects` table with role greater or equal to `Owner`).
+* the user has a role greater or equal to `Owner` for a resource hierarchically superior to the one being checked (e.g. 
+ again using projects as an example, if there is no record in the `users_projects` table, but there is a record for the 
+ given user with role greater or equal to `Owner` for the organization that owns the project).
```

Comment:
This would require our module to be aware of relations and query for related resources (and thus ask other modules).

I would recommend keeping it as much separated as possible and self-sufficient, not importing anything.

Reply:
Same comment as above https://github.com/Vizzuality/marxan-cloud/pull/621#discussion_r744575478

Reply:
Rewriting this too according to https://github.com/Vizzuality/marxan-cloud/pull/621#discussion_r744681803